File: 1_arrays_and_strings/1.2/js/main.js
/**
 * Question: 1.2
 * Name: Check Permutation
 * Description: Given two strings, write a method to decide if one is a permutation of the other.
 * Solution: The strings must pass two inspections to be considered a permutation of each other:
 *  1) They should be the same length.
 *  2) Their sorted contents should be equal.
 */


/**
 * Helper function that returns the sorted version of a string.
 */
const sorted = s => {
  return s
    .split('')
    .sort()
    .join('');
};


/**
 * Determines if two strings are a permutation of each other using two conditions:
 * 1) The strings must be the same length.
 * 2) Their sorted contents must be the same.
 */
const isPermutation = (s1, s2) => {
  if (s1.length !== s2.length) {
    return false;
  }

  return (sorted(s1) === sorted(s2));
};



console.log(isPermutation('care', 'race'));         // true
console.log(isPermutation('care', 'rack'));         // false
console.log(isPermutation('care', 'care'));         // true
console.log(isPermutation('care', 'care bear'));    // false
console.log(isPermutation('care', 'car'));          // false

File: 1_arrays_and_strings/1.2/python/main.py
"""
Question: 1.2
Description: Given two strings, write a method to decide if one is a permutation of the other.
"""


def is_permutation(s1, s2):
    """
    Solution: The strings must pass two inspections to be considered a permutation of each other:
      1) They should be the same length.
      2) Their sorted contents should be equal.
    """
    if len(s1) != len(s2):
        return False

    return ''.join(sorted(s1)) == ''.join(sorted(s2))


#
# Examples
#
print(is_permutation('care', 'race'))         # True
print(is_permutation('care', 'rack'))         # False
print(is_permutation('care', 'care'))         # True
print(is_permutation('care', 'care bear'))    # False
print(is_permutation('care', 'car'))          # False
